### DISCUSSION TITLE
v5.4.1 Release - Discussion, Feedback, Ideas

### DISCUSSION THREAD

**Discussion by dewenni** [dewenni/ESP_Buderus_KM271#132] [Announcements]
## what's new

This is a major update with some new features and some redesign under the hood.
The Framework is updated to the new Arduino v3 based on ESP-IDF v5.1+
Several modifications have been made to achieve better performance and stability.

### Performance and stability improvements

Various improvements have been made to increase performance and stability. In particular, the communication between ESP and the web server has been redesigned and improved. The protocol was changed from SSE (Server-Sent-Events) to Websockets.
The maximum number of client connections is monitored and limited to the number of 2 active connections.

### Ethernet Support W5500

There is a new support for W5500 Ethernet Modules. This brings the possibility to use ethernet connection in parallel to WiFi connection.
This could be useful in situations where the WiFi connection is not available or stable.

For mor details, please have a look at the [documentation](https://github.com/dewenni/ESP_Buderus_KM271?tab=readme-ov-file#optional-ethernet-module-w5500) 

There is also an expansion module in the pipeline that fits perfectly on the KM271 WiFi board.  
https://www.tindie.com/products/the78mole/km271-wifi-ethernet-extension

![image](https://github.com/user-attachments/assets/27f5e32f-b981-47e9-bdc8-9ac579b3e1f5)

### Solar Support FM244

Support for optional Buderus Solar Module FM244 added. You can activate this in the same way as the heating circuits and hot-water.
Values are send by mqtt and are also available in the webUI.

### System Logger

The logging function has been extended. The mode can now also be switched to ‘System log’. In this case, the serial monitor is redirected to the log and you can see the messages that are normally only visible when you are connected via cable.

### new HC1/HC2 Control Elements

There are new HC1/HC2 control elements on the web UI

- HC1 Day Temperature / HC2 Day Temperature
- HC1 Night Temperature / HC2 Night Temperature
- HC1 Holiday Temperature / HC2 Holiday Temperature
- HC1 Switch-On Temperature / HC2 Switch-On Temperature

### additional Status Information in the web UI

There are some additional information in the status Tab of the web UI

- Status information about Ethernet connection
- Status information about KM271 connection
- Status information about MQTT connection

### BREAKING CHANGES!!!

> [!IMPORTANT]   
> In the past, there was a separate setting for the static IP. Now we have two possible interfaces and therefore the static IP setting is also duplicated and is located together with the WiFi and Ethernet settings. If you previously used static IP, you will need to adjust your settings.
>
>**I would recommend deactivating the static IP before the update!**

## changelog

- update to espressif/arduino-esp32 V3.07
- change AsyncTCP library to [mathieucarbou/AsyncTCP](https://github.com/mathieucarbou/AsyncTCP) v3.2.14
- update ESPAsyncWebServer [mathieucarbou/ESPAsyncWebServer](https://github.com/mathieucarbou/ESPAsyncWebServer) v3.3.22
- change communication from ServerSentEvents to websockets
- optimization and reduction of the data sent from the ESP to the web server
- remove "uptime" from MQTT discovery, because it floods the log #108
- fix typo "Ferhler" in OTA Dialog #108
- fix name of exported config.json file
- clean unused webUI texts
- direct apply of the settings after import of config.json
- update Descriptions for Ethernet and Exhaust Sensor in README.md
- add Task Watchdog to automatically reboot if ESP is getting stuck
- add Solar FM244 support #64
- add status icons for WiFi and Ethernet
- add "Mode: SystemLog" to the WebUI Logger function that shows the ESP_LOG messages
- add Status information about Ethernet connection
- add Status information about KM271 connection
- add Status information about MQTT connection
- add control for HC1 Day Temperature / HC2 Day Temperature
- add control for HC1 Night Temperature / HC2 Night Temperature to webUI
- add control for HC1 Holiday Temperature / HC2 Holiday Temperature to webUI
- add control for HC1 Switch-On Temperature / HC2 Switch-On Temperature to webUI

**Comment by dewenni** [comment]
# v5.1.0

## what's new

This is a minor update version, but it is based on a major update 5.0.0
Please see here what´s changed in [release v5.0.0](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v5.0.0)

> [!IMPORTANT]   
> unfortunately there was a bug in the MQTT handling in v5.0.0 which led to commands not being accepted correctly. Therefore please update to this version if you are already using 5.0.0!

### reminder Ethernet Extention

I would also like to point out again that since V5 it is possible to connect the ESP via Ethernet.  
Daniel already has a great expansion board designed for this, which you are welcome to register for.  
https://www.tindie.com/products/the78mole/km271-wifi-ethernet-extension


![](https://github.com/dewenni/ESP_Buderus_KM271/blob/main/Doc/w5500_extention.png)


### new functions

- now the active Page is highlighted in the navigation bar.
- due to the increased number of possible used GPIO´s - there is a check whether the GPIOs are valid and individual GPIOs are not configured multiple times

## changelog

- bugfix incorrect handling of received mqtt payload #131
- bugfix incorrect heap usage value
- change mqtt client-id back to WiFi-Hostname #128
- no restart of the ESP if the WiFi connection cannot be established but there is an Ethernet connection.
- interval for renewed mqtt connection attempt increases with each attempt by a further 10s (max. 50s). The ESP restarts after 5 failed attempts
- change previous used Libraries for DS18B20 sensors to [mathieucarbou/MycilaDS18](https://github.com/mathieucarbou/MycilaDS18)
- add function to check whether the configured GPIOs are valid and have not been assigned twice
- highlight active page in sidebar navigation
- update Readme.md

**Reply by Sprudelverduenner** [reply]
Hi dewenni,

habe soeben gesehen, dass es eine neue Version 5.xx gibt. Ich wollte auf die aktuelle Version updaten.

Egal welche Version von 5.xx ich als OTA Update einspielen möchte: es kommt bei ca. 80% Fortschritt bei mir immer zu der Fehlermeldung, dass der Speicher nicht ausreichend groß wäre.

Hast Du eine Idee ?

Danke und frohes Fest,
Sprudelverduenner

**Reply by dewenni** [reply]
@Sprudelverduenner 
Leider kann ich dir hier auch keinen entsprechenden Hinweis geben. Der Speicher ist leider mit ca. 82% schon ziemlich voll. Aber es sollte eigentlich noch ausreichend Platz sein. Bei mir und vermutlich den meisten anderen, hat das Update geklappt.
Vermutlich musst du es nochmal mit einem Kabel flashen.
Ich würde dir das Update auch empfehlen, da ich im Vergleich zur v4 einiges an der Stabilität verbessert habe.

**Reply by Sprudelverduenner** [reply]
Danke für deine Antwort.
Wenn ich mit einem Kabel drüber flashe : bleiben die Einstellungen erhalten?

**Reply by kalleick** [reply]
Exportiere Dir die Einstellungen vorher über den Menüpunkt "Tools", Konfiguration import/export

**Reply by dewenni** [reply]
ja genau, du kannst dir die Einstellungen vorher sichern.
Beim späteren upload bitte prüfen das die Datei auch `config.json` heißt und nicht z.B. `config (1).json` oder so.
Derzeit ändere ich den Dateinamen beim Upload noch nicht automatisch. Das habe ich im nächsten Release aber auch geändert. Dann ist es egal wie die Datei beim upload heißt, sie wird nach dem upload immer in `config.json` umbenannt.

**Comment by DonCanDo** [comment]
Meine Platine ist vorhin nach gut 4 5 Tagen auch abrupt abgeschmiert. Der einzige suspekte Wert den ich fand war Heap usage
ESP Heap usage | 267801.0 | %

**Reply by dewenni** [reply]
Kannst du noch ein paar Infos dazu liefern?
Kannst du das "abschmieren" zeitlich mit einer Aktion in Verbindung bringen? Beispielsweise als du das WebUI geöffnet hast?
- Hat der ESP dann neu gestartet?
- Was steht denn im WebUI als "Restart Reason"?

Der hohe Wert der "Heap Usage" war noch ein Umrechnungsproblem, welches ich in 5.1.0 behoben habe. Das war im Grund nur ein "Anzeigefehler" Mach dazu bitte nochmal ein Update.

**Comment by DonCanDo** [comment]
Guten Morgen,

bislang viel mir nichts auf außer, dass es genau auf der Minute passierte. 

Ich konnte im Log leider nichts finden, weil ich den Strom trennen musste.
Der Eintrag mit Home Assistant stopped liegt bestimmt mit meinem Neustart in Zusammenhang.

Mit dem Heap ja da hast du Recht im HAOS wird es richtig übertragen.

Ich melde mich mal, wenn es wieder aufgetreten ist ich glaube hier kann man jetzt noch sehr wenige Schlüsse ziehen.
![Unbenannt](https://github.com/user-attachments/assets/692fe8ca-d309-487e-a410-b21eedda52bc)

**Comment by vovanmm661905** [comment]
Hello, dewenni. In my humble opinion, the parameter "HC2_MixingValue" should be int8, not uint8. And the units of measurement are not percentages, but steps. When the actual temperature matches the required value, "HC2_MixingValue" becomes 0. A negative value indicates the direction of closing the valve, and a positive value indicates opening.

**Reply by dewenni** [reply]
yes, that fits to what @solutionfound observed here: https://github.com/dewenni/ESP_Buderus_KM271/issues/133#issuecomment-2562389234
I will change that with the next release.
Thanks!

**Comment by dewenni** [comment]
# v5.3.0

## what's new

### direct GitHub Update 🎉🎉🎉

This update introduces an exciting new feature: You can now update to the latest version directly from GitHub in the WebUI – no need to download the .bin file manually anymore!

Simply click on the version info in the bottom left corner, or the new Button in the Tool Section and a dialog will open. If a new version is available, you can initiate the update right there. It will automatically download and install the latest release from GitHub.

![ota-2](https://raw.githubusercontent.com/dewenni/ESP_Buderus_KM271/main/Doc/github_ota.gif)

> [!TIP]
> Maybe it is necessary to clean your browser cache after the update, to be sure that everything works well!

### encrypted Passwords

Passwords are now better protected and stored in encrypted form in the config.json
When updating, the existing passwords are automatically encrypted and saved again.

effected config elements:

- WiFi Password
- Web Authentication Password
- MQTT Password
- Pushover API_Key
- Pushover User_Key

> [!CAUTION]
> As the passwords are stored in encrypted form after this update, it is no longer possible to switch to an older version without re-entering the passwords after the downgrade

## changelog

- [UPDATE] Arduino core 3.1.1 based on IDF 5.3.2.241224
- [UPDATE] mathieucarbou/AsyncTCP @ 3.3.2
- [UPDATE] mathieucarbou/ESPAsyncWebServer @ 3.6.0
- [FEATURE] new feature to update directly from GitHub
- [FEATURE] Home Assistant: `state_class: measurement` added for Logamatic Status values #138
- [IMPROVE] Improved behavior when the restart button is pressed immediately after a change in the settings.
- [IMPROVE] Passwords are better protected and are stored in encrypted form

**Comment by d5coupe** [comment]
Hi, I am very new to this.  I flashed Daniel's board v0/0/6 with your latest FW 5.3.1, I am able to enter setup and define settings.  I click reboot, then installed into my Logamatic. I am unable to connect to 192.168.4.1.  The board is blinking green.  How do I know whats wrong?  How do I proceed?  Thank you so much!

**Reply by dewenni** [reply]
Have you setup your WiFi Credentials (SSID, Password)?
If yes, then the board should connect to your WiFi Network.
You have to find out what IP-Adress the board gets from your router and then use this IP-Adress to connect to the WebUI.

The IP-Adress 192.168.4.1 is only used in the Setup Mode when no other WiFi Connection ist configured.

**Reply by d5coupe** [reply]
OK, got it running now!  How do I connect to MQTT for Home assistant?  Not sure if I need a server...
Next, my wifi signal and Rssi is not registering on "System" tab.  Is this normal?
Also, is it possible to add an option to show temperature in degrees Fahrenheit? ((0°C × 9/5) + 32 = 32°F)
Many many thanks!

**Reply by the78mole** [reply]
For Home Assistant, there is an MQTT server addon called Mosquitto. You can easily install it and then enter the corresponding data in the KM271. As far as I know, dewennis firmware also supports Home Assistant Discovery, which means all entities are automatically registered in Home Assistant, allowing you to integrate them into a dashboard, create automations, and whatever else you can think of.

**Comment by dewenni** [comment]
# v5.4.0

## what's new

this brings different improvements and some new function.

### WebUI-Demo

For a first impression of the functions and the WebUI, a limited demo is also available.  
This can be accessed via the following link: [WebUI-DEMO](https://dewenni.github.io/ESP_Buderus_KM271/)

### Changed condition for Setup-Mode

Setup mode is now activated when the ESP is restarted **5** times.
A maximum of 5 seconds may elapse after each restart.

Example: restart 1/5 - wait 2s - restart 2/5 - wait 2s - restart 3/5 - wait 2s - restart 4/5 - wait 2s - restart /5/5 => Setup-Mode

### Extended Logger

The Logger source is now split into "System" and "Logamatic" log source.
System and Logamatic is recorded in parallel. You can switch between this two sources without deleting the buffer.
But the buffer will be deleted if you change the log filter for the Logamatic log.

### Flexible Oil-Meter Configuration

For the Hardware Oil-Meter you can now configure the "pulses per litre" and a "debouncing time"
For the virtual Oil-Meter you can noe define a offset value to correct the calculated value to your needs.

## changelog

- [UPDATE]  ESP32Async/AsyncTCP @ 3.3.5
- [UPDATE]  ESP32Async/ESPAsyncWebServer @ 3.7.0
- [UPDATE]  update ArduinoJSON @ 7.4.0
- [CHANGE]  Basic WebUI functions were outsourced to a separate [EspWebUI](https://github.com/dewenni/EspWebUI) library
- [IMPROVE] rework internal logging functions
- [IMPROVE] Added validation for IP input fields in the setup area to avoid wrong syntax
- [IMPROVE] Conversion of the storage of the oil counter from EEPROM to NVS
- [IMPROVE] change "Double-Reset-Detection" to "Multi-Reset-Detection" now it needs 5 restarts within 5s timeout to enter Setup-Mode
- [FEATURE] Add a [WebUI-DEMO](https://dewenni.github.io/ESP_Buderus_KM271/) under github-pages
- [FEATURE] For the Hardware Oil-Meter you can now configure the "pulses per litre" and a "debouncing time" 
#148
- [FEATURE] For the virtual Oil-Meter you can noe define a offset value to correct the calculated value to your needs. 
#147

**Comment by dewenni** [comment]
# v5.4.1

## what's new

this is a minor bugfix version.

## changelog

- [UPDATE]  ESP32Async/ESPAsyncWebServer @ 3.7.6
- [UPDATE]  update ArduinoJSON @ 7.4.1
- [UPDATE]  dewenni/EspWebUI @ 0.0.7
- [BUGFIX]  Recalculate the virtual oil meter value when any dependent parameter changes 
(#147).
- [BUGFIX]  Ensure system info is sent regardless of Logamatic initialization 
(#152)
- [IMPROVE] Add FAQ section to README and reference it in the issue template
- [IMPROVE] Preserve hardware oil meter value across restarts
- [IMPROVE] Automatically refresh Logger when the Logger Page is opened
- [IMPROVE] Colorize logger messages depending on log level

**Comment by langer88os** [comment]
I would like to have a Rest api to get the Data as json. This allows me to better integrate it into my logging System. 
What do you think about it?

**Reply by dewenni** [reply]
Could be an option. Please publish this as a new idea in a separate post in the discussion area.
thanks
